thingsvision/core/extraction/helpers.py

```diff
@@ -1,8 +1,193 @@
+
+        CustomExtractor.show_model = show_model
+        CustomExtractor.forward = forward
+        CustomExtractor.flatten_acts = flatten_acts
+
+    custom_extractor = CustomExtractor(
+        model_name=model_name, 
+        pretrained=pretrained,
+        device=device, 
+        model_path=model_path, 
+        model=model, 
+        preprocess=preprocess
+    )
+
+    return custom_extractor
+
+
+
+def create_model_extractor(
+    model: Any,
+    device: str, 
+    preprocess: Any = None,
+    backend: str = "pt",
+    forward_fn: Callable = None,
+) -> Any:
+    """
+    Creates a class for extracting activations from a given model (PyTorch or TensorFlow).
+
+    Parameters:
+    -----------
+    model: Any
+        The model from which activations will be extracted.
+    device: str
+        The device on which the model is loaded.
+    preprocess: Any
+        The preprocessing function to be applied to the input images (default: None).
+    backend: str
+        The backend of the model. Either "pt" for PyTorch or "tf" for TensorFlow.
+    forward_fn: Callable
+        In case your model requires more complicated forward passes than simply using model(img),
+        you can pass a custom forward function here. The function must have the following signature:
+        
+        forward_fn(self, img, module_name) -> activations
+        
+        and calls to the model have to be made on the self.model attribute.
+
+    Returns:
+    --------
+    extractor: Any
+        The custom extractor class.
+    """
+    backend_mixin = PyTorchMixin if backend == "pt" else TensorFlowMixin
+
+    class ModelExtractor(BaseExtractor, backend_mixin):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+    if forward_fn:
+        ModelExtractor.forward = forward_fn
+
+    model_extractor = ModelExtractor(
+        model_name="custom", 
+        model_path=None, 
+        device=device, 
+        model=model, 
+        preprocess=preprocess
+    )
+
+    return model_extractor
+
+def get_extractor(
+    model_name: str,
+    pretrained: bool,
+    device: str,
+    source: str,
+    model_path: str = None,
+    model_parameters: Dict[str, str] = None,
+) -> Any:
+    model_args = {
+        "model_name": model_name,
+        "model_path": model_path,
+        "device": device,
+        "pretrained": pretrained,
+        "model_parameters": model_parameters,
+    }
+
+    """Get a model extractor from <source> library."""
+    if source == "torchvision":
+        return TorchvisionExtractor(**model_args)
+    elif source == "timm":
+        return TimmExtractor(**model_args)
+    elif source == "keras":
+        return KerasExtractor(**model_args)
+    elif source == "custom":
+        return create_custom_extractor(**model_args)
+    else:
+        raise ValueError(
+            f"\nCould not find {source} library.\nChoose a different source.\n"
+        )
+
+
+def get_extractor_from_model(
```

Comment:
I think there is no test for this? But there is also a separate issue for creating extractors from model instances (#88 ) so maybe the test could also be done with that

Reply:
Let's add test in conjunction with resolving issue #88?